components/raftstore/src/store/fsm/peer.rs

```diff
@@ -2631,6 +2621,9 @@ where
         // We can't destroy a peer which is handling snapshot.
         assert!(!self.fsm.peer.is_handling_snapshot());
 
+        // No need to wait for the apply anymore.
+        self.finish_unsafe_recovery_wait_apply();
```

Comment:
it may not be waiting for unsafe recovery

Reply:
Thank you for catching this.